### PR TITLE
Fix index deletion order

### PR DESCRIPTION
**Pull request by AntonEbel**
- consider the `_warm` infix when extracting index number
- remove code duplication

/nocl

/jenkins-pr-deps https://github.com/Graylog2/graylog-plugin-enterprise/pull/6267

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/periodical/IndexRetentionThread.java [hunk 1]
```diff
@@ -145,6 +145,6 @@ public int getInitialDelaySeconds() {
 
     @Override
     public int getPeriodSeconds() {
-        return (int) MINUTES.toSeconds(5);
+        return (int) MINUTES.toSeconds(1);
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/periodical/IndexRetentionThread.java:148]: Did you change that on purpose? Or is it a leftover from testing?

File: graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java [hunk 2]
```diff
@@ -188,6 +181,14 @@ public Optional<Integer> extractIndexNumber(final String indexName) {
         }
     }
 
+    private int indexPrefixLength(String indexName) {
+        int length = config.indexPrefix().length() + 1;
+        if (indexName.contains(WARM_INDEX_INFIX)) {
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java:186]:
```suggestion
        if (indexName.contains("_" + WARM_INDEX_INFIX)) {
```
We could make this match more precise.
Also, could we add a validation that prevents the user from creating index sets with the name `warm_`?
Or are we sure this doesn't cause any problems?

Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java:186]: I just tested it. using my change fixes it for an index set with prefixes `warm_`
```
green open warm__warm_0 wM-hSUntTqqy8Go1WrOqeg 4 0 196 0  201kb  201kb
green open warm__2      KF3H7fBUTNaNezMJJDlIZA 5 0  13 0 13.6kb 13.6kb
green open warm__1      k9Fcst3QQIONOIZ3FPTY0A 5 0 959 0  1.1mb  1.1mb
```

File: graylog2-server/src/main/java/org/graylog2/periodical/IndexRetentionThread.java (hunk 1 above)
Comment (AntonEbel) [reply graylog2-server/src/main/java/org/graylog2/periodical/IndexRetentionThread.java]: thanks for catching that

File: graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java (hunk 2 above)
Comment (AntonEbel) [reply graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java]: I added also the validation for the keyword `warm`

File: graylog2-server/src/main/java/org/graylog2/indexer/IndexSetValidator.java [hunk 3]
```diff
@@ -134,10 +134,14 @@ public Violation validateRefreshInterval(Duration readableDuration) {
 
     @Nullable
     private Violation validatePrefix(IndexSetConfig newConfig) {
+        if (newConfig.indexPrefix().contains(MongoIndexSet.KEYWORD_WARM)) {
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/indexer/IndexSetValidator.java:137]:
```suggestion
        if (newConfig.indexPrefix().equals(MongoIndexSet.KEYWORD_WARM)) {
```
I think this is too strict. Allowing `warm` in a word won't hurt us

**Pull request merged**